### PR TITLE
[bridge] Extract WorkspaceInstanceController

### PR DESCRIPTION
**Pull request by geropl**
## Description
Preparations in order to work on #6770:
 - make telepresence work again
 - fix a bug in how we configure preview envs ([commit](https://github.com/gitpod-io/gitpod/pull/13866/commits/97733893fe9580e455b5c115b87c1d0d5626726a))
 - extract `WorkspaceInstanceController` from `Bridge` (code move only, zero function change)

## Related Issue(s)
Context: #6770

## How to test
 - open [a workspace on this PR](https://github.com/gitpod-io/gitpod/pull/13866) and see that the `ws-manager-bridge` deployment run without issues
 - check logs of `ws-manager-bridge` and not that there are no errors
 - start a workspace and notice how everything works as expected.

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Comment by geropl** [comment]
While werft is not picking up the changes from [commit](https://github.com/gitpod-io/gitpod/pull/13866/commits/97733893fe9580e455b5c115b87c1d0d5626726a) regarding the preview envs, fixing it manually in the ConfigMaps works nicely:
image

File: components/ws-manager-bridge/src/container-module.ts [hunk 1]
```diff
@@ -91,4 +92,6 @@ export const containerModule = new ContainerModule((bind) => {
     bind(DebugApp).toSelf().inSingletonScope();
 
     bind(Client).toDynamicValue(getExperimentsClientForBackend).inSingletonScope();
+
+    bind(WorkspaceInstanceController).to(WorkspaceInstanceControllerImpl).inTransientScope();
```
Comment (easyCZ) [inline components/ws-manager-bridge/src/container-module.ts:96]: What does transient mean?

File: components/ws-manager-bridge/src/workspace-instance-controller.ts [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        controllerMaxDisconnectSeconds: number,
+    ): void;
+
+    controlNotStoppedAppClusterManagedInstanceTimeouts(
+        parentCtx: TraceContext,
+        runningInstances: RunningWorkspaceInfo[],
+        installation: string,
+    ): Promise<void>;
+
+    onInstanceStopped(ctx: TraceContext, ownerUserID: string, instance: WorkspaceInstance): Promise<void>;
```
Comment (easyCZ) [inline components/ws-manager-bridge/src/workspace-instance-controller.ts:39]:
```suggestion
    onStopped(ctx: TraceContext, ownerUserID: string, instance: WorkspaceInstance): Promise<void>;
```
The interface is already a WorksapceInstanceController, no need to duplicate Instance in the name again.

File: components/ws-manager-bridge/src/workspace-instance-controller.ts [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+export interface WorkspaceInstanceController {
+    start(
+        installation: string,
+        clientProvider: ClientProvider,
+        controllerIntervalSeconds: number,
+        controllerMaxDisconnectSeconds: number,
+    ): void;
+
+    controlNotStoppedAppClusterManagedInstanceTimeouts(
```
Comment (easyCZ) [inline components/ws-manager-bridge/src/workspace-instance-controller.ts:33]: This is a mouthful. How about 
```suggestion
    handleZombies(
```
or something similar. We can introduce a noun which corresponds to this Java like expression to improve mind association

File: components/ws-manager-bridge/src/workspace-instance-controller.ts [hunk 4]
```diff
@@ -0,0 +1,308 @@
+/**
```
Comment (easyCZ) [inline components/ws-manager-bridge/src/workspace-instance-controller.ts:1]: I'm assuming this is just a lift from the original file, are there any significant changes to this file?

File: components/ws-manager-bridge/src/workspace-instance-controller.ts (hunk 3 above)
Comment (geropl) [reply components/ws-manager-bridge/src/workspace-instance-controller.ts:33]: I'd like to keep the explicit name, as it avoids confusion, and fits into the pattern of a similarly named function in the same class.

File: components/ws-manager-bridge/src/workspace-instance-controller.ts (hunk 2 above)
Comment (geropl) [reply components/ws-manager-bridge/src/workspace-instance-controller.ts:39]: Good point, missed that opportunity, will change.

File: components/ws-manager-bridge/src/container-module.ts (hunk 1 above)
Comment (geropl) [reply components/ws-manager-bridge/src/container-module.ts:96]: A new instance is created for every request. Will explain a bit in the documentation.

**Review by easyCZ — APPROVED** [review]
LGTM, thanks for the clarifications (and explaining transient)

**Comment by geropl** [comment]
/unhold

Applied the mentioned suggestions, rebased and re-tested: :heavy_check_mark:

**Pull request merged**